### ISSUE TITLE
help command crashes Nu in Windows Terminal

### ISSUE THREAD

**Issue by rgwood** [nushell/nushell#1491]
Running the `help` command in Windows Terminal crashes Nu with the following error:

```
Error: Io(Os { code: 87, kind: Other, message: "The parameter is incorrect." })
```

**Details**
1. `help` produces its usual output before crashing.
1. I cannot reproduce this in cmd.exe, Windows Powershell, or on macOS. It only seems to happen in Windows Terminal.
1. `help` usually crashes the first time it's invoked, but sometimes it takes 2 or 3 tries.

**Configuration**
 - OS: Windows 10 version 1909
 - Nu: 0.11.0 installed via Cargo
 - Terminal: Windows Terminal 0.9.433.0

**Comment by jonathandturner** [comment]
@rgwood - does this happen with `help` with no arguments?

**Comment by rgwood** [comment]
Yes, sorry for not clarifying. `help where` and `help ls` seem fine.

**Comment by jonathandturner** [comment]
@rgwood - I wonder if it's happening because of how long the message is.

If you run something like `type <long_text_file.txt>` do you also get a crash?  Just wondering if printing out long messages is enough to trigger it, or if there's something specific to `help`

**Comment by slavoutich** [comment]
(Using nushell-0.11.1, alacritty) 

This seems to be connected with some output before it, say:
```
C:\Users\slavo> nu
Welcome to Nushell 0.11.1 (type 'help' for more info)
C:\Users\slavo> pwd
C:\Users\slavo
C:\Users\slavo> pwd
C:\Users\slavo
C:\Users\slavo> pwd
C:\Users\slavo
C:\Users\slavo> ls
────┬─────────────────────┬─────────┬─────────┬──────────────
 #  │ name                │ type    │ size    │ modified
────┼─────────────────────┼─────────┼─────────┼──────────────
(output)
────┴─────────────────────┴─────────┴─────────┴──────────────
C:\Users\slavo> pwd
C:\Users\slavo
Error: Io(Os { code: 87, kind: Other, message: "The parameter is incorrect." })
```
(first pwds are succeeding, but after it dies)

**Comment by rgwood** [comment]
Interesting. For me it crashes the very first time I run `pwd`.

`type <long_text_file.txt>` does _not_ crash (tested with the README.md from this repo).

**Comment by jonathandturner** [comment]
I was just able to repro this also.

**Comment by jonathandturner** [comment]
For those investigating: we've seen this pattern in the past and managed to track it down to issues in rustyline:

eg)
* https://github.com/kkawakam/rustyline/issues/305
* https://github.com/kkawakam/rustyline/issues/249

I don't want to point fingers while we're still figuring out what is happening, but if we're trying to get a repro in place, we may want to see if they're similar to the past issues and go from there.

**Comment by fdncred** [comment]
I just got this issue in a session i started like this: `nu.exe --loglevel debug`.
```
Apps/NuShell/nu_latest
❯  DEBUG rustyline                    > Vi insert: SelfInsert(1, 'p')
 DEBUG rustyline                    > Changeset::insert(0, 'p')
 DEBUG rustyline                    > old layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: true, cursor: Position { col: 2, row: 1 }, end: Position { col: 2, row: 1 } }
 DEBUG rustyline                    > new layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: truApps/NuShell/nu_latest
❯ p DEBUG rustyline                    > Vi insert: SelfInsert(1, 'w')
 DEBUG rustyline                    > Changeset::insert(1, 'w')
 DEBUG rustyline                    > old layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: true, cursor: Position { col: 3, row: 1 }, end: Position { col: 5, row: 1 } }
 DEBUG rustyline                    > new layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: truApps/NuShell/nu_latest
❯ pw DEBUG rustyline                    > Vi insert: SelfInsert(1, 'd')
 DEBUG rustyline                    > Changeset::insert(2, 'd')
 DEBUG rustyline                    > old layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: true, cursor: Position { col: 4, row: 1 }, end: Position { col: 5, row: 1 } }
 DEBUG rustyline                    > new layout: Layout { prompt_size: Position { col: 2, row: 1 }, default_prompt: truApps/NuShell/nu_latest
❯ pwd DEBUG rustyline                    > Vi insert: AcceptLine
 DEBUG rustyline                    > Changeset::begin
 DEBUG rustyline                    > Changeset::end

 DEBUG nu_cli::cli                  > === Parsed ===
 DEBUG nu_cli::cli                  > LitePipeline {
    commands: [
        LiteCommand {
            name: Spanned {
                span: Span {
                    start: 0,
                    end: 3,
                },
                item: "pwd",
            },
            args: [],
        },
    ],
}
 DEBUG starship::config             > STARSHIP_CONFIG is not set
 DEBUG starship::config             > Using default config path: C:\Users\username\.config/starship.toml
 DEBUG starship::config             > Unable to read config file content:
The system cannot find the file specified. (os error 2)
 DEBUG starship::config             > Config found for "git_status":
Some(Table({"disabled": Boolean(true)}))
 DEBUG starship::modules::directory > Current directory: "C:\\Apps\\NuShell\\nu_latest"
 DEBUG starship::config             > Config found for "git_branch":
Some(Table({"symbol": String("📙 ")}))
 DEBUG starship::config             > Config found for "git_branch":
Some(Table({"symbol": String("📙 ")}))
 DEBUG rustyline                    > ANSI colors already enabled
 DEBUG rustyline                    > Changeset::delete(0, "")
 DEBUG rustyline                    > Changeset::insert_str(0, "")
C:\Apps\NuShell\nu_latest DEBUG rustyline                    > initial cursor location: 25, 29

Error: Io(Os { code: 87, kind: Other, message: "The parameter is incorrect." })
```
I typed `pwd<enter_key>` and nu.exe exited.

**Comment by jonathandturner** [comment]
I dug around and I think I found where it's happening, though I currently don't have a fix.

Lines like https://github.com/nushell/nushell/blob/master/crates/nu-cli/src/commands/autoview.rs#L122 mean that we're sending large strings into the output buffer.  Rustyline will pick up this output buffer since it's likely still writing to the screen and then panic because i/o is dirty at that point.

I tried manually flushing these, but I suspect part of the problem is that the output is also on the same line that rustyline wants to be on.  If we switch them to `outln!` instead, the crash goes away.

We can work around the `help` crasher, but it's possible other large strings will cause issues, too.

cc @thegedge who did the original fix (https://github.com/nushell/nushell/pull/1466)

**Comment by jonathandturner** [comment]
More investigations here: https://github.com/kkawakam/rustyline/issues/359

**Comment by jonathandturner** [comment]
Should be fixed by #1581

**Comment by jonathandturner** [comment]
This should now be fixed in master
